agent/config/runtime_test.go

```diff
@@ -5996,23 +5999,18 @@ func TestFullConfig(t *testing.T) {
 		t.Run(format, func(t *testing.T) {
 			// parse the flags since this is the only way we can set the
 			// DevMode flag
-			var flags Flags
+			var flags BuilderOpts
 			fs := flag.NewFlagSet("", flag.ContinueOnError)
 			AddFlags(fs, &flags)
 			if err := fs.Parse(flagSrc); err != nil {
 				t.Fatalf("ParseFlags: %s", err)
 			}
 
```

Comment:
Should we add a check like `require.Len(t, fs.Args(), 0)` here as well? Since we no longer error for that in `NewBuilder`.